packages/marshaler.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package packages
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type MarshallerOption func(packages *Packages) error
+
+func MarshalJSON(pkgs *Packages) ([]byte, error) {
+	return json.MarshalIndent(pkgs, " ", " ")
+}
+
+func UnmarshalJSON(content []byte, pkgs *Packages, options ...MarshallerOption) error {
+	err := json.Unmarshal(content, pkgs)
+	if err != nil {
+		return err
+	}
+
+	for i := range *pkgs {
+		err = (*pkgs)[i].setRuntimeFields()
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, opt := range options {
+		err = opt(pkgs)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ResolveBasePaths(packagesPath ...string) MarshallerOption {
```

Comment:
I find this a bit hacky, and not sure if we are going to need this. For cases where the package is available locally, we use the current local indexers. For the packages in the remote index, we unmarshal the index JSON, and we don't have the packages locally.

I would remove this if not needed.

Reply:
Removed all options.